### PR TITLE
Move preview UI code to Procedure Editor

### PR DESCRIPTION
**Pull request by makiam**
Move preview UI code to Procedure Editor where it only used. Preview window now follow procedure editor

### CODE REVIEW

File: ArtOfIllusion/src/artofillusion/procedural/ProcedureOwner.java [hunk 1]
```diff
@@ -22,20 +24,14 @@ define the user interface for editing it. */
   
   /** Create an object which displays a preview of the procedure. */
   
-  default Object getPreview(ProcedureEditor editor)
+  default MaterialPreviewer getPreview()
```
Comment (Sailsman63) [inline ArtOfIllusion/src/artofillusion/procedural/ProcedureOwner.java:27]: Changing the signature here technically breaks compatibility at both a source and binary level. While updating the core plugins is simple, have you tested that this does not break TaPD, which builds of the procedural system?

**Comment by makiam** [comment]
TAPD plugin use its own modules system not tied to aoi

вт, 7 мар. 2023 г., 10:50 Sailsman63 ***@***.***>:

**Comment by peteihis** [comment]
To me this and #301 seem like very wrong things to do. 

The thinking very much through the entire AoI has been that each object should do itself everything that is related to it and I think there is a good reason behind it. That of course leads to some level of redundancy in the code but what is going to happen when somebody add something new that needs an entirely different logic?

Let's not start breakin the basic principles that the software has been built on.

**Comment by makiam** [comment]
>>  the entire AoI has been that each object should do itself everything that is related to it
Totally agree with this. And this way the ProceduralXXX objects should NOT create UI parts for ProceduralEditor and should NOT control disposal of UI (via disposePreview).

**Comment by Sailsman63** [comment]
I don't think that this one has the same sort of issues as #301. It does, however, rest on two assumptions:

1. All procedure editors will want their preview windows to be laid out in the same way.
2. All types of procedures will be able to be effectively previewed through a `MaterialPreviewer` Widget.

The first I am unsure about. The second I have more grave suspicions: better, I think, to accept a generic Widget here.

**Comment by makiam** [comment]
Once ProcedureEditor changes some in it's editing object it triggers ProcedureOwner updatePreview(previewer)  anв inside it usually calls MaterialPreviewer render(). So if pass previewer as Widget we need extra cast and we have no confidence that proper object passed to update

**Pull request closed without merging**